scripts/git/update_to_llvm_syncpoint.py

```diff
@@ -0,0 +1,248 @@
+      "--tensorflow_commit",
+      "--tf_commit",
+      help=
+      f"Update TensorFlow to this commit, or a named option: {COMMIT_OPTIONS}",
+      default=INTEGRATE_COMMIT)
+  parser.add_argument(
+      "--validate",
+      help="Validate that the selected commits all match the LLVM commit",
+      type=utils.str2bool,
+      nargs="?",
+      default=True,
+  )
+
+  parser.add_argument("--update_build_files",
+                      help="Updates the IREE LLVM build files from TensorFlow.",
+                      type=utils.str2bool,
+                      nargs="?",
+                      default=True)
+  args = parser.parse_args()
+
+  # Default repo path.
+  if args.repo is None:
+    args.repo = utils.find_git_toplevel()
+
+  # Set some defaults.
+  if not args.tensorflow:
+    args.tensorflow = os.path.join(args.repo, "third_party", "tensorflow")
+  if not args.llvm:
+    args.llvm = os.path.join(args.repo, "third_party", "llvm-project")
+
+  return args
+
+
+def main(args):
+  print("IREE handy-dandy-LLVM-submodule-updater at your service...")
+  print("  IREE Path :", args.repo)
+  print("  LLVM Path :", args.llvm)
+  print("  TensorFlow Path :", args.tensorflow)
+  print("  Update Build files:", args.update_build_files)
+  current_llvm_commit = get_commit(args.llvm)
+  current_tensorflow_commit = get_commit(args.tensorflow)
+
+  print("Current Commits: llvm =", current_llvm_commit, "tensorflow =",
+        current_tensorflow_commit)
+
+  # Update TensorFlow
+  new_tf_commit = find_new_tf_commit(args.tensorflow, current_llvm_commit,
+                                     args.tensorflow_commit)
+  print("\n*** Updating TensorFlow to", new_tf_commit, "***")
+  utils.execute(["git", "checkout", new_tf_commit], cwd=args.tensorflow)
+  stage_path(args.repo, args.tensorflow)
+
+  validate_tf_commit(current_llvm_commit,
+                     args.tensorflow,
+                     exit_on_failure=args.validate)
+
+  if args.update_build_files:
+    print("\n*** Updating BUILD.bazel files ***")
+    update_build_files_from_tensorflow(args.repo, args.tensorflow)
+
+  # Export SUBMODULE_VERSIONS.
+  print()  # Add line break.
+  submodule_versions.export_versions(args.repo)
+
+
+def get_commit(path, rev="HEAD"):
+  return utils.execute(["git", "rev-parse", rev],
+                       cwd=path,
+                       silent=True,
+                       capture_output=True,
+                       universal_newlines=True).strip()
+
+
+def find_new_tf_commit(tensorflow_path, llvm_commit, tf_commit):
+  utils.execute(["git", "fetch"], cwd=tensorflow_path)
+
+  if tf_commit not in COMMIT_OPTIONS:
+    return get_commit(tensorflow_path, rev=tf_commit)
+
+  if tf_commit == KEEP_COMMIT:
+    return get_commit(tensorflow_path)
+
+  if tf_commit == REMOTE_HEAD_COMMIT:
+    return get_commit(tensorflow_path, "origin/master")
+
+  tf_integrate_commits = utils.execute([
+      "git", "log", "--first-parent", "--format=%H", "-S", llvm_commit,
```

Comment:
Is there a longer option name for `-S`? https://blog.danlew.net/2019/02/19/git-log-s/ explains it, but I don't see it on https://git-scm.com/docs/git-log. This page: https://git-scm.com/book/en/v2/Git-Tools-Searching and some other sources call it a "pickaxe search". Might be worth a comment or link to a reference page explaining the intent here.

(I was skeptical about the methodology here and the assert below for `len(tf_integrate_commits) <= 2`, since there could be commits with "extra fixes for LLVM commit $SHA", but it appears that this only searches code changes, not commit messages)

Reply:
Yeah I looked for a longer name and didn't find one. I'll add more documentation here

https://git-scm.com/docs/git-log#Documentation/git-log.txt--Sltstringgt